pkg/machinery/labels/validate.go

```diff
@@ -130,3 +130,28 @@ func ValidateLabelValue(value string) error {
 
 	return nil
 }
+
+// ValidateTaints validates that a set of taints is correctly defined.
+func ValidateTaints(taints map[string]string) error {
+	var multiErr *multierror.Error
+
+	keys := maps.Keys(taints)
+	slices.Sort(keys)
+
+	for _, k := range keys {
+		if err := ValidateQualifiedName(k); err != nil {
+			multiErr = multierror.Append(multiErr, err)
+		}
+
+		val, effect, found := strings.Cut(taints[k], ":")
+		if found && effect != "NoSchedule" && effect != "PreferNoSchedule" && effect != "NoExecute" {
```

Comment:
doesn't k8s supports taints other than this :thinking: ?

Reply:
cool, maybe we keep this in `constants`? and then filter on a slice?

Reply:
I dunno, both works for me.

Reply:
Updated the code.